packages/server/api/src/app/core/security/utils.ts

```diff
@@ -0,0 +1,53 @@
+import { AuthorizationType, RouteKind, ProjectResourceType } from "@activepieces/server-shared"
+import { ActivepiecesError, ErrorCode, isNil } from "@activepieces/shared"
+import { FastifyRequest } from "fastify"
+import { requestUtils } from "../request/request-utils"
+
+export const securityUtils = {
+    async getProjectIdFromRequest(request: FastifyRequest): Promise<string> {
+        const security = request.routeOptions.config?.security
```

Comment:
The check if it's project authorization should be done where you decide the authorization, so coming to this function the code already know there is project authorization.

This can be enforced instead of sending request: FastifyRequest, you could send the `request.authorization.project` directly intoi the parameter 

Then the function would be small enough and only used in one place that check the project authorization, so if it's only used in one place, maybe it's good to move it there.